internal/services/network/network_watcher_flow_log_resource.go

```diff
@@ -72,11 +72,14 @@ func resourceNetworkWatcherFlowLog() *pluginsdk.Resource {
 				ValidateFunc: validate.NetworkWatcherFlowLogName,
 			},
 
-			"network_security_group_id": {
-				Type:         pluginsdk.TypeString,
-				Required:     true,
-				ForceNew:     true,
-				ValidateFunc: networksecuritygroups.ValidateNetworkSecurityGroupID,
+			"target_resource_id": {
+				Type:     pluginsdk.TypeString,
+				ForceNew: true,
```

Comment:
This wants to be set to false in the `!features.FivePointOhBeta` block so that users can migrate to using this property without recreating the resource